### PR TITLE
MRG: Adding existence check to the mne write raw bids.

### PR DESCRIPTION
**Pull request by adam2392**
PR Description
--------------

Fixes: #330 

Added existence check or `overwrite=True`. Do I need to still log something in `whats_new.rst`?

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [x] All comments resolved
- [x] This is not your own PR
- [ ] All CIs are happy
- [x] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [x] PR description includes phrase "closes <#issue-number>"
- [x] Commit history does not contain any merge commits

### CODE REVIEW

**Comment by jasmainak** [comment]
yep, update `whats_new.rst` and also can you add a test?

**Comment by adam2392** [comment]
> yep, update `whats_new.rst` and also can you add a test?

Done. What do you think about adding a default for the `Authors` keyword? I believe this would make any mne_bids `write_raw_bids` then error free using the: https://bids-standard.github.io/bids-validator/ ?

Else, there's always an error saying authors need to be in the dataset_description.json. 

I guess on the other hand, it could be a good thing to force authors to literally update the dataset_description explicitly?

**Comment by jasmainak** [comment]
humm indeed, I would have made the "authors" field a positional argument so users are forced to provide that. On the other hand, I'm also fine making it write something like "MNE-BIDS" so people know it has been automatically converted by MNE-BIDS and then edited.

**Comment by jasmainak** [comment]
@adam2392 please set the PR to MRG once you are ready. It looks good to me so far.

**Comment by jasmainak** [comment]
CIs are failing and it looks legitimate

**Comment by adam2392** [comment]
> CIs are failing and it looks legitimate

Just to confirm, it doesn't look like an issue on our end right?

**Comment by jasmainak** [comment]
Let's see if this fixes it: https://github.com/mne-tools/mne-bids/pull/336

**Comment by jasmainak** [comment]
@adam2392 can you rebase this PR? Should fix the CI and then it should be good.

**Comment by adam2392** [comment]
@jasmainak is the travis CI failing because of a rebase issue? I am pretty sure I rebased correctly this time.

I checked the mainline repo, and it looks like CIs worked, but this PR's had something to do w/ bids_validator version on a MAC build?...

File: doc/whats_new.rst [hunk 1]
```diff
@@ -22,6 +22,10 @@ Current
 Changelog
 ~~~~~~~~~
 
+- Add existence check for :func:`write_raw_bids` before :func:`make_dataset_description` is called, by `Adam Li`_ (`#331 <https://github.com/mne-tools/mne-bids/pull/331>`_)
+- Standardized `bids_root` and `output_path` arguments in :func:`read_raw_bids`, :func:`write_raw_bids`, and :func:`make_bids_folder` to just `bids_root`, by `Adam Li`_ (`#303 <https://github.com/mne-tools/mne-bids/pull/303>`_)
+- Added landmark argument to :func:`write_anat` to pass landmark location directly for deface, by `Alex Rockhill`_ (`#292 <https://github.com/mne-tools/mne-bids/pull/292>`_)
```
Comment (jasmainak) [inline doc/whats_new.rst]: something went wrong here, this is also below

File: doc/whats_new.rst [hunk 2]
```diff
@@ -22,6 +22,10 @@ Current
 Changelog
 ~~~~~~~~~
 
+- Add existence check for :func:`write_raw_bids` before :func:`make_dataset_description` is called, by `Adam Li`_ (`#331 <https://github.com/mne-tools/mne-bids/pull/331>`_)
```
Comment (jasmainak) [inline doc/whats_new.rst]: normally we add the new changes at the bottom not at the top, so the whats_new file reads chronologically and not reverse-chronologically

File: mne_bids/write.py [hunk 3]
```diff
@@ -1164,7 +1168,10 @@ def write_raw_bids(raw, bids_basename, bids_root, events_data=None,
     if events is not None and len(events) > 0 and not emptyroom:
         _events_tsv(events, raw, events_fname, event_id, overwrite, verbose)
 
-    make_dataset_description(bids_root, name=" ", verbose=verbose)
+    dataset_description_fpath = op.join(bids_root, "dataset_description.json")
+    if not op.exists(dataset_description_fpath) or overwrite:
+        make_dataset_description(bids_root, name=" ", verbose=verbose)
```
Comment (jasmainak) [inline mne_bids/write.py]: humm I didn't realize this was already being created in write_raw_bids. Can you update the test from the PR of @sappelhoff : https://github.com/mne-tools/mne-bids/pull/342/files

File: doc/whats_new.rst [hunk 4]
```diff
@@ -30,6 +30,7 @@ Changelog
 - Add ability for :func:`mne_bids.get_head_mri_trans` to read fiducial points from fif data without applying maxfilter, by `Maximilien Chaumon`_ (`#291 <https://github.com/mne-tools/mne-bids/pull/291>`_)
 - Added landmark argument to :func:`write_anat` to pass landmark location directly for deface, by `Alex Rockhill`_ (`#292 <https://github.com/mne-tools/mne-bids/pull/292>`_)
 - Standardized `bids_root` and `output_path` arguments in :func:`read_raw_bids`, :func:`write_raw_bids`, and :func:`make_bids_folder` to just `bids_root`, by `Adam Li`_ (`#303 <https://github.com/mne-tools/mne-bids/pull/303>`_)
+- Add existence check for :func:`write_raw_bids` before :func:`make_dataset_description` is called, by `Adam Li`_ (`#331 <https://github.com/mne-tools/mne-bids/pull/331>`_)
```
Comment (jasmainak) [inline doc/whats_new.rst]: I think should go in the section on bugfixes.

**Comment by jasmainak** [comment]
Other than my minor nitpick, it's good to merge. Thanks for your patience @adam2392 !

**Pull request merged**